### PR TITLE
Optimize the 'error' signature

### PR DESCRIPTION
**Pull request by zsxwing**
As per #80

This will break the source compatibility. E.g., `Observable.error[Int](new Exception(msg)).firstOrElse(10).toBlocking.single` won't be compiled.

Hold off until we prepare to release 0.24.0.

### CODE REVIEW

**Comment by samuelgruetter** [comment]
I would not make this change for the reasons I explained in https://github.com/ReactiveX/RxScala/issues/80#issuecomment-66110613

File: src/test/scala/rx/lang/scala/ObservableTest.scala [hunk 1]
```diff
@@ -428,4 +428,15 @@ class ObservableTests extends JUnitSuite {
     assertEquals(expectedMap3, m3.toBlocking.single)
   }
 
+  @Ignore // Don't run it. Only test if it can be compiled
+  def testErrorSignature() {
+    val o: Observable[Int] = Observable.error(new RuntimeException("Oops")).map(v => v)
+    println(o.toBlocking.single)
+
+    val x = 1
+    println(x + (Observable.error(new RuntimeException("Oops")).toBlocking.single: Int))
```
Comment (zsxwing) [inline src/test/scala/rx/lang/scala/ObservableTest.scala:437]: @samuelgruetter Just revisited this one and found `error[T]` also needs to specify `: Int`. Now I cannot find a case that `error[T]` is better than `error[Nothing]`. So how about changing to `error[Nothing]`?

Comment (headinthebox) [inline src/test/scala/rx/lang/scala/ObservableTest.scala:437]: Yes !

Comment (samuelgruetter) [inline src/test/scala/rx/lang/scala/ObservableTest.scala:437]: Some people might prefer `Observable.error[Int](...)` over a `... : Int` ascription... ;-)
But I guess such examples will be very rare, only occur in constructed examples, so I'd be ok with `def error(exception: Throwable): Observable[Nothing]` (but note that it's a breaking change).

**Pull request closed without merging**